### PR TITLE
Unify server setup and call log configuration (#695)

### PR DESCRIPTION
**Pull request by yfei1**
This commits is a hotfix for the stackdriver bug #569. It also enables the log trace config in Open Match. One thing to note is #695 is based on the new helm chart that we will be using in v.7 so I made [a small twist](https://github.com/yfei1/open-match/blob/55e8793898011471469c2b1d155fe98df14c6b4f/install/helm/open-match/values.yaml#L138) in `values.yaml` file to make the go template works for v0.6. This workaround will go away in v.7.

### CODE REVIEW

File: install/helm/open-match/values.yaml [hunk 1]
```diff
@@ -135,7 +135,7 @@ openmatch:
       matchmaker_config.yaml:
         logging:
           level: debug
-          format: text
+          format: "{{ternary \"stackdriver\" \"text\" .Values.openmatch.telemetry.stackdriver.enabled}}"
```
Comment (sawagh) [inline install/helm/open-match/values.yaml]: How to interpret this? how does this get set?

File: internal/app/appmain.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		logger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Fatalf("cannot construct server.")
+	}
+
+	if err := bindService(p, cfg); err != nil {
+		logger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Fatalf("failed to bind backend service.")
```
Comment (sawagh) [inline internal/app/appmain.go]: add the service name here than just backend service.

File: install/helm/open-match/values.yaml (hunk 1 above)
Comment (yfei1) [reply install/helm/open-match/values.yaml]: This calls the ternary function defined in the gotpl package - http://masterminds.github.io/sprig/defaults.html#ternary. However, helm does not allow templating in the values.yaml file so what we does in v.6 is to quote the templates and pass the strings to the `tpl` function defined in om-configmap.yaml, which makes the `values.yaml` file cumbersome and hard to modify - I have to quote and escape the template to make it works.

This restriction will go away in v0.7 as we introduced the new helm chart.

File: internal/app/appmain.go (hunk 2 above)
Comment (yfei1) [reply internal/app/appmain.go]: Done.

**Pull request merged**